gym/envs/registration.py

```diff
@@ -17,22 +17,49 @@
 
 from gym import error, logger
 
-# This format is true today, but it's *not* an official spec.
-# [username/](env-name)-v(version)    env-name is group 1, version is group 2
-#
-# 2016-10-31: We're experimentally expanding the environment ID format
-# to include an optional username.
-env_id_re: re.Pattern = re.compile(
-    r"^(?:(?P<namespace>[\w:-]+)\/)?(?P<name>[\w:.-]+)-v(?P<version>\d+)$"
-)
 
+def check_env_id(env_str: str) -> dict:
+    """Check environment ID string format.
 
-def env_id_from_parts(namespace: str, name: str, version: Union[int, str]) -> str:
+    This format is true today, but it's *not* an official spec.
+    [username/](env-name)-v(version)    env-name is group 1, version is group 2
+
+    2016-10-31: We're experimentally expanding the environment ID format
+    to include an optional username.
+    """
+    # First we try the strict pattern with the version number
+    pattern = re.compile(
```

Comment:
Could you use the following regular expression:
```
^(?:(?P<namespace>[\w:-]+)\/)?(?:(?P<name>[\w:.-]+?))(?:-v(?P<version>\d+))?$
```
so we can have this logic all in one expression. This way we can just check if the capture group `version` is None or not.

All I changed was to make the name match group non-greedy and make the version group optional (greedy).

Reply:
Right, way better this way, thanks for the hint! :pray: